crates/bevy_reflect/src/set.rs

```diff
@@ -0,0 +1,555 @@
+    /// If the set did have this value present, `false` is returned.
+    fn insert_boxed(&mut self, value: Box<dyn Reflect>) -> bool;
+
+    /// Removes a value from the set.
+    ///
+    /// If the set did not have this value present, `true` is returned.
+    /// If the set did have this value present, `false` is returned.
+    fn remove(&mut self, value: &dyn Reflect) -> bool;
+
+    /// Checks if the given value is contained in the set
+    fn contains(&self, value: &dyn Reflect) -> bool;
+}
+
+/// A container for compile-time set info.
+#[derive(Clone, Debug)]
+pub struct SetInfo {
+    type_path: TypePathTable,
+    type_id: TypeId,
+    value_type_path: TypePathTable,
+    value_type_id: TypeId,
+    #[cfg(feature = "documentation")]
+    docs: Option<&'static str>,
+}
+
+impl SetInfo {
+    /// Create a new [`SetInfo`].
+    pub fn new<TSet: Set + TypePath, TValue: Reflect + TypePath>() -> Self {
+        Self {
+            type_path: TypePathTable::of::<TSet>(),
+            type_id: TypeId::of::<TSet>(),
+            value_type_path: TypePathTable::of::<TValue>(),
+            value_type_id: TypeId::of::<TValue>(),
+            #[cfg(feature = "documentation")]
+            docs: None,
+        }
+    }
+
+    /// Sets the docstring for this set.
+    #[cfg(feature = "documentation")]
+    pub fn with_docs(self, docs: Option<&'static str>) -> Self {
+        Self { docs, ..self }
+    }
+
+    /// A representation of the type path of the set.
+    ///
+    /// Provides dynamic access to all methods on [`TypePath`].
+    pub fn type_path_table(&self) -> &TypePathTable {
+        &self.type_path
+    }
+
+    /// The [stable, full type path] of the set.
+    ///
+    /// Use [`type_path_table`] if you need access to the other methods on [`TypePath`].
+    ///
+    /// [stable, full type path]: TypePath
+    /// [`type_path_table`]: Self::type_path_table
+    pub fn type_path(&self) -> &'static str {
+        self.type_path_table().path()
+    }
+
+    /// The [`TypeId`] of the set.
+    pub fn type_id(&self) -> TypeId {
+        self.type_id
+    }
+
+    /// Check if the given type matches the set type.
+    pub fn is<T: Any>(&self) -> bool {
+        TypeId::of::<T>() == self.type_id
+    }
+
+    /// A representation of the type path of the value type.
+    ///
+    /// Provides dynamic access to all methods on [`TypePath`].
+    pub fn value_type_path_table(&self) -> &TypePathTable {
+        &self.value_type_path
+    }
+
+    /// The [`TypeId`] of the value.
+    pub fn value_type_id(&self) -> TypeId {
+        self.value_type_id
+    }
+
+    /// Check if the given type matches the value type.
+    pub fn value_is<T: Any>(&self) -> bool {
+        TypeId::of::<T>() == self.value_type_id
+    }
+
+    /// The docstring of this set, if any.
+    #[cfg(feature = "documentation")]
+    pub fn docs(&self) -> Option<&'static str> {
+        self.docs
+    }
+}
+
+/// An ordered mapping between reflected values.
```

Comment:
This comment is not very clear or useful.

Reply:
I'm not really sure what to put there either tbh. But it was still just the copy-paste from the `Map` equivalent so I at least changed that a little bit.